template.go

```diff
@@ -4,16 +4,16 @@ package cli
 // cli.go uses text/template to render templates. You can
 // render custom help text by setting this variable.
 var AppHelpTemplate = `NAME:
-   {{.Name}}{{if .Usage}} - {{.Usage}}{{end}}
+   {{$v := offset .Name 6}}{{wrap .Name 3}}{{if .Usage}} - {{wrap .Usage $v}}{{end}}
 
 USAGE:
-   {{if .UsageText}}{{.UsageText}}{{else}}{{.HelpName}} {{if .VisibleFlags}}[global options]{{end}}{{if .Commands}} command [command options]{{end}} {{if .ArgsUsage}}{{.ArgsUsage}}{{else}}[arguments...]{{end}}{{end}}{{if .Version}}{{if not .HideVersion}}
+   {{if .UsageText}}{{wrap .UsageText 3}}{{else}}{{.HelpName}} {{if .VisibleFlags}}[global options]{{end}}{{if .Commands}} command [command options]{{end}} {{if .ArgsUsage}}{{.ArgsUsage}}{{else}}[arguments...]{{end}}{{end}}{{if .Version}}{{if not .HideVersion}}
```

Comment:
❓ &nbsp; _(blocking question)_ what do the 6 and 3 here represent???

Reply:
They are arguments to the functions `wap` and `offset`, in the funcMap applied to the template.

`wrap` takes a string and an offset to indent wrapped lines, and returns a wrapped string (the second line onwards are all indented by the offset. `3` is the indent level used in the template.

`offset` returns the length of a string plus a given number, and is used to calculate an offset for the description field of the app name. I hope my terrible ascii art can explain where the `6` comes from (the sum of the two hard-coded `3`'s in this part of the template):

```
NAME:
   <appname> - <description blah blah blah...
               we want wrapped lines to line
               up here>
^-^^-------^^-^
 |    |      |
 |    |      another hardcoded 3
 |    |
 |   length of app name
 |
hardcoded 3

```

Reply:
I'm satisfied with this answer.